src/credentials/sso/diskCache.ts

```diff
@@ -0,0 +1,102 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as fs from 'fs'
+import { SsoCache } from './ssoCache'
+import { SsoClientRegistration } from './ssoClientRegistration'
+import { join } from 'path'
+import { homedir } from 'os'
+import { SsoAccessToken } from './ssoAccessToken'
+import { getSHA1StringHash } from '../../shared/utilities/textUtilities'
+import { getLogger } from '../../shared/logger'
+
+export class DiskCache implements SsoCache {
+    private cacheDir: string
+
+    constructor(cacheDir: string = join(homedir(), '.aws', 'sso', 'cache')) {
+        this.cacheDir = cacheDir
+    }
+    // Treat the token or client registration as expired if within 15 minutes of expiration.
+    private TOKEN_EXPIRATION_BUFFER_MS = 900000
+
+    public loadClientRegistration(ssoRegion: string): SsoClientRegistration | undefined {
+        if (!this.registrationExists(ssoRegion)) {
+            return undefined
+        }
+        try {
+            const registration = JSON.parse(fs.readFileSync(this.registrationCache(ssoRegion)).toString())
+            if (registration && this.isNotExpired(registration)) {
+                return registration
+            }
+        } catch (error) {
+            getLogger().error(error)
+        }
+
+        return undefined
+    }
+
+    public saveClientRegistration(ssoRegion: string, registration: SsoClientRegistration): void {
+        fs.mkdirSync(this.cacheDir, { recursive: true })
+        // According to the Spec: 'SSO Login Token Flow': the file permissions for the cached client registration
+        // must be 0600 (owner read and write)
+        fs.writeFileSync(this.registrationCache(ssoRegion), JSON.stringify(registration), {
+            mode: 0o600,
+        })
+    }
+    public invalidateClientRegistration(ssoRegion: string): void {
+        if (this.registrationExists(ssoRegion)) {
+            fs.unlinkSync(this.registrationCache(ssoRegion))
+        }
+    }
+    public loadAccessToken(ssoUrl: string): SsoAccessToken | undefined {
+        if (!this.tokenExists(ssoUrl)) {
+            return undefined
+        }
+        try {
+            const accessToken = JSON.parse(fs.readFileSync(this.accessTokenCache(ssoUrl)).toString())
+            if (accessToken && this.isNotExpired(accessToken)) {
+                return accessToken
+            }
+        } catch (error) {
+            getLogger().error(error)
```

Comment:
Nit: I'm assuming that this error would come from the `JSON.parse`? Is there a chance that any private information could be outputted by this?

Reply:
The error would be a syntax error.  I ran this with a bad JSON object and didn't see any info passed with with the error.